### PR TITLE
Fixed "vttablet --help" to show correct command line flags

### PR DESCRIPTION
**Pull request by rasikakale**
fixes: vitessio#4932  

I fixed the vttablet --help error which gave flags that were not applicable by doing the following fixes:

- Created `testhandler` and `staticauthserver` package within `mysql` package

- `testhandler` contains the following files that I moved from `mysql` into this package:
    - `conn_test.go`
    - `query_test.go`
    - `replication_test.go`
    - `testhandler.go` (contains TestHandler type originally part of `mysql/query_benchmark_test.go`)

-  `staticauthserver` contains the following files also moved from `mysql` into this package:
    - `auth_server_static_test.go`
    - `auth_server_static.go`

- These files were moved from the `mysql `package to the `endtoend` package:
    - `handshake_test.go`
    - `server_test.go`
    - `auth_server_clientcert_test.go`

- `static_user_data.go` file added to the `mysql` package that contains the StaticUserData type originally in `staticauthserver/auth_server_static.go`

- I added prefixes of the appropriate package name in front of functions that are called outside of the package they are located in
    - e.g. A function `RegisterAuthServerImpl()` is defined in `mysql/auth_server.go` but called in `staticauthserver/auth_server_static.go`. The appropriate fix would be `mysql.RegisterAuthServerImpl()
`

For the record, the auth options do not show up for `vttablet --help`

Signed-off-by: Rasika Kale rasika@planetscale.com

### CODE REVIEW

File: examples/local/env.sh [hunk 1]
```diff
@@ -77,3 +77,5 @@ else
     mkdir -p "${VTDATAROOT}/tmp"
     mkdir -p "${VTDATAROOT}/etcd"
 fi
+
```
Comment (deepthi) [inline examples/local/env.sh]: revert unnecessary whitespace change to this file

File: go/mysql/auth_server.go [hunk 2]
```diff
@@ -79,20 +79,20 @@ type AuthServer interface {
 	Negotiate(c *Conn, user string, remoteAddr net.Addr) (Getter, error)
 }
 
-// authServers is a registry of AuthServer implementations.
-var authServers = make(map[string]AuthServer)
+// AauthServers is a registry of AuthServer implementations.
```
Comment (deepthi) [inline go/mysql/auth_server.go]: Is this a typo? AuthServers, not AauthServers?

File: go/mysql/auth_server.go [hunk 3]
```diff
@@ -79,20 +79,20 @@ type AuthServer interface {
...
+var AauthServers = make(map[string]AuthServer)
 
 // RegisterAuthServerImpl registers an implementations of AuthServer.
-func RegisterAuthServerImpl(name string, authServer AuthServer) {
-	if _, ok := authServers[name]; ok {
+func RegisterAuthServerImpl(name string, AauthServer AuthServer) {
```
Comment (deepthi) [inline go/mysql/auth_server.go]: same comment as above

File: go/mysql/auth_server.go [hunk 4]
```diff
@@ -79,20 +79,20 @@ type AuthServer interface {
 	Negotiate(c *Conn, user string, remoteAddr net.Addr) (Getter, error)
 }
 
-// authServers is a registry of AuthServer implementations.
-var authServers = make(map[string]AuthServer)
+// AauthServers is a registry of AuthServer implementations.
+var AauthServers = make(map[string]AuthServer)
```
Comment (sougou) [inline go/mysql/auth_server.go]: `AauthServers`->`AuthServers`

**Comment by sougou** [comment]
I'll close this for now. We  can revisit this later.
